megamek/src/megamek/utilities/ai/UtilityPathRankerCostFunction.java

```diff
@@ -0,0 +1,1204 @@
+        }
+
+
+        private record QuadrantParameters(int startX, int startY, int endX, int endY) {
+        }
+
+        /**
+         * Remove all strategic goals on the quadrant of the given coordinates
+         * @param coords The coordinates to check
+         */
+        @SuppressWarnings("unused")
+        public void removeAllStrategicGoalsOnCoordsQuadrant(Coords coords) {
+            QuadrantParameters quadrant = getQuadrantParameters(coords);
+            for (int i = quadrant.startX(); i < quadrant.endX(); i++) {
+                for (int j = quadrant.startY(); j < quadrant.endY(); j++) {
+                    strategicGoals.remove(new Coords(i, j));
+                }
+            }
+        }
+
+        /**
+         * Initialize the strategic goals for the board
+         * @param board The board to initialize the goals on
+         * @param quadrantWidth The width of the quadrants
+         * @param quadrantHeight The height of the quadrants
+         */
+        public void initializeStrategicGoals(Board board, int quadrantWidth, int quadrantHeight) {
+            strategicGoals.clear();
+            this.quadrantWidth = quadrantWidth;
+            this.quadrantHeight = quadrantHeight;
+
+            int boardWidth = board.getWidth();
+            int boardHeight = board.getHeight();
+
+            // Calculate extra space and offsets to center the quadrants
+            int extraX = boardWidth % quadrantWidth;
+            int extraY = boardHeight % quadrantHeight;
+            offsetX = extraX / 2;
+            offsetY = extraY / 2;
+
+            // Iterate over each quadrant using the offsets
+            for (int i = 0; i < (boardWidth - offsetX); i += quadrantWidth) {
+                for (int j = 0; j < (boardHeight - offsetY); j += quadrantHeight) {
+                    int startX = offsetX + i;
+                    int startY = offsetY + j;
+                    int endX = Math.min(startX + quadrantWidth, boardWidth);
+                    int endY = Math.min(startY + quadrantHeight, boardHeight);
+
+                    var xMidPoint = (startX + endX) / 2;
+                    var yMidPoint = (startY + endY) / 2;
+                    for (var coords : new Coords(xMidPoint, yMidPoint).allAtDistanceOrLess(3)) {
+                        var hex = board.getHex(coords);
+                        if (hex == null || hex.isClearHex() && hasNoHazards(hex)) {
+                            addStrategicGoal(coords);
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        private static final Set<Integer> HAZARDS = new HashSet<>(Arrays.asList(Terrains.FIRE,
+            Terrains.MAGMA,
+            Terrains.ICE,
+            Terrains.WATER,
+            Terrains.BUILDING,
+            Terrains.BRIDGE,
+            Terrains.BLACK_ICE,
+            Terrains.SNOW,
+            Terrains.SWAMP,
+            Terrains.MUD,
+            Terrains.TUNDRA));
```

Comment:
Could re-use the previously defined set.

Reply:
We could, I will move it to inside the hex then.